src/sys/unix/uds/listener.rs

```diff
@@ -1,148 +1,84 @@
-                .and_then(|_| syscall!(fcntl(socket, libc::F_SETFD, libc::FD_CLOEXEC)).map(|_| s))
-        });
-
-        socket.and_then(|s| Ok((s, SocketAddr::from_parts(sockaddr, socklen))))
-    }
-
-    pub(crate) fn bind(path: &Path) -> io::Result<UnixListener> {
-        let socket = new_socket(libc::AF_UNIX, libc::SOCK_STREAM)?;
-        let (sockaddr, socklen) = socket_addr(path)?;
-        let sockaddr = &sockaddr as *const libc::sockaddr_un as *const libc::sockaddr;
-
-        syscall!(bind(socket, sockaddr, socklen))
-            .and_then(|_| syscall!(listen(socket, 1024)))
-            .map_err(|err| {
-                // Close the socket if we hit an error, ignoring the error from
-                // closing since we can't pass back two errors.
-                let _ = unsafe { libc::close(socket) };
-                err
-            })
-            .map(|_| unsafe { UnixListener::from_raw_fd(socket) })
-    }
-
-    pub fn from_std(inner: net::UnixListener) -> UnixListener {
-        UnixListener { inner }
-    }
-
-    pub(crate) fn local_addr(&self) -> io::Result<SocketAddr> {
-        SocketAddr::new(|sockaddr, socklen| {
-            syscall!(getsockname(self.inner.as_raw_fd(), sockaddr, socklen))
-        })
-    }
-
-    /// Returns the value of the `SO_ERROR` option.
-    pub(crate) fn take_error(&self) -> io::Result<Option<io::Error>> {
-        self.inner.take_error()
-    }
-}
-
-impl event::Source for UnixListener {
-    fn register(
-        &mut self,
-        registry: &Registry,
-        token: Token,
-        interests: Interest,
-    ) -> io::Result<()> {
-        SourceFd(&self.as_raw_fd()).register(registry, token, interests)
-    }
-
-    fn reregister(
-        &mut self,
-        registry: &Registry,
-        token: Token,
-        interests: Interest,
-    ) -> io::Result<()> {
-        SourceFd(&self.as_raw_fd()).reregister(registry, token, interests)
-    }
-
-    fn deregister(&mut self, registry: &Registry) -> io::Result<()> {
-        SourceFd(&self.as_raw_fd()).deregister(registry)
-    }
-}
-
-impl AsRawFd for UnixListener {
-    fn as_raw_fd(&self) -> RawFd {
-        self.inner.as_raw_fd()
-    }
-}
-
-impl FromRawFd for UnixListener {
-    unsafe fn from_raw_fd(fd: RawFd) -> UnixListener {
-        UnixListener::new(net::UnixListener::from_raw_fd(fd))
-    }
+        .map(|socket| unsafe { net::UnixStream::from_raw_fd(socket) })
+    };
+
+    #[cfg(any(
+        target_os = "ios",
+        target_os = "macos",
+        target_os = "netbsd",
+        target_os = "solaris"
+    ))]
+    let socket = syscall!(accept(
+        listener.as_raw_fd(),
+        &mut sockaddr as *mut libc::sockaddr_un as *mut libc::sockaddr,
+        &mut socklen,
+    ))
+    .and_then(|socket| {
+        // Ensure the socket is closed if either of the `fcntl` calls
+        // error below.
+        let s = unsafe { net::UnixStream::from_raw_fd(socket) };
+        syscall!(fcntl(socket, libc::F_SETFL, libc::O_NONBLOCK))
```

Comment:
Remove this in a separate change to reflect `tcp::accept` where we know the new FD inherits file status?

Reply:
Created #1250.